src/validator/AdminValidator.cpp

```diff
@@ -8,83 +8,97 @@
 #include "common/charset/Charset.h"
-#include "util/SchemaUtil.h"
+#include "common/interface/gen-cpp2/meta_types.h"
 #include "parser/MaintainSentences.h"
-#include "service/GraphFlags.h"
 #include "planner/Admin.h"
 #include "planner/Query.h"
+#include "service/GraphFlags.h"
+#include "util/SchemaUtil.h"
 
 namespace nebula {
 namespace graph {
 Status CreateSpaceValidator::validateImpl() {
     auto sentence = static_cast<CreateSpaceSentence*>(sentence_);
     ifNotExist_ = sentence->isIfNotExist();
     auto status = Status::OK();
-    spaceDesc_ = meta::SpaceDesc();
-    spaceDesc_.spaceName_ = std::move(*(sentence->spaceName()));
+    spaceDesc_.space_name = std::move(*(sentence->spaceName()));
     StatusOr<std::string> retStatusOr;
     std::string result;
     auto* charsetInfo = qctx_->getCharsetInfo();
     for (auto &item : sentence->getOpts()) {
         switch (item->getOptType()) {
             case SpaceOptItem::PARTITION_NUM: {
-                spaceDesc_.partNum_ = item->getPartitionNum();
-                if (spaceDesc_.partNum_ <= 0) {
+                spaceDesc_.partition_num = item->getPartitionNum();
+                if (spaceDesc_.partition_num <= 0) {
                     return Status::Error("Partition_num value should be greater than zero");
                 }
                 break;
             }
             case SpaceOptItem::REPLICA_FACTOR: {
-                spaceDesc_.replicaFactor_ = item->getReplicaFactor();
-                if (spaceDesc_.replicaFactor_ <= 0) {
+                spaceDesc_.replica_factor = item->getReplicaFactor();
+                if (spaceDesc_.replica_factor <= 0) {
                     return Status::Error("Replica_factor value should be greater than zero");
                 }
                 break;
             }
-            case SpaceOptItem::VID_SIZE: {
-                spaceDesc_.vidSize_ = item->getVidSize();
-                if (spaceDesc_.vidSize_ < 0 ||
-                        spaceDesc_.vidSize_ > std::numeric_limits<int32_t>::max()) {
-                    return Status::Error("Vid_size value is incorrect");
+            case SpaceOptItem::VID_TYPE: {
+                auto typeDef = item->getVidType();
+                if (typeDef.type != meta::cpp2::PropertyType::INT64 &&
+                        typeDef.type != meta::cpp2::PropertyType::FIXED_STRING) {
+                    std::stringstream ss;
+                    ss << "Only support FIXED_STRING or INT64 vid type, but was given "
+                       << meta::cpp2::_PropertyType_VALUES_TO_NAMES.at(typeDef.type);
+                    return Status::Error(ss.str());
+                }
+                spaceDesc_.vid_type = typeDef.type;
+
+                if (typeDef.type == meta::cpp2::PropertyType::INT64) {
+                    spaceDesc_.vid_size = 8;
+                } else {
+                    if (typeDef.typeLen <= 0 ||
+                        typeDef.typeLen > std::numeric_limits<int16_t>::max()) {
```

Comment:
The thrift interface defines the typeLen is int32, why here limit int16?

Reply:
It was defined `int16` in `ColumnTypeDef`.

Reply:
I am so sorry, it’s my fault, I define the wrong type.  the parser has checked the range. need to modify the type to int32_t.

Reply:
ok.